leveldb/shardedLeveldbSerial_test.go

```diff
@@ -0,0 +1,425 @@
+package leveldb_test
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-storage-go/leveldb"
+	"github.com/multiversx/mx-chain-storage-go/testscommon"
+	"github.com/multiversx/mx-chain-storage-go/types"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	_1Mil = 1_000_000
+	_2Mil = 2_000_000
+	_4Mil = 4_000_000
+	_8Mil = 8_000_000
+	_10k  = 10_000
+	_100k = 100_000
+
+	numShards = 4
+
+	singleID  = "single"
+	shardedID = "sharded"
+)
+
+func TestNewShardedPersister(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil id provider", func(t *testing.T) {
+		t.Parallel()
+
+		dir := t.TempDir()
+		db, err := leveldb.NewShardedPersister(dir, 2, 10, 10, nil)
+		require.Nil(t, db)
+		require.Equal(t, leveldb.ErrNilIDProvider, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		dir := t.TempDir()
+		db, err := leveldb.NewShardedPersister(dir, 2, 10, 10, &testscommon.ShardIDProviderStub{})
+		require.NotNil(t, db)
+		require.Nil(t, err)
+	})
+}
+
+func TestShardedPersister_Operations(t *testing.T) {
+	t.Parallel()
+
+	dir := t.TempDir()
+	db, err := createPersister(dir, shardedID)
+	require.Nil(t, err)
+
+	_ = db.Put([]byte("aaa"), []byte("aaaval"))
+	_ = db.Put([]byte("aab"), []byte("aabval"))
+	_ = db.Put([]byte("aac"), []byte("aacval"))
+
+	err = db.Close()
+	require.Nil(t, err)
+
+	db2, err := createPersister(dir, shardedID)
+	require.Nil(t, err)
+
+	_, err = db2.Get([]byte("aaa"))
+	require.Nil(t, err)
+
+	_, err = db2.Get([]byte("aab"))
+	require.Nil(t, err)
+
+	_, err = db2.Get([]byte("aac"))
+	require.Nil(t, err)
+
+	require.Nil(t, db2.Has([]byte("aaa")))
+
+	err = db2.Remove([]byte("aaa"))
+	require.Nil(t, err)
+
+	require.NotNil(t, db2.Has([]byte("aaa")))
+
+	numKeys := 0
+	db2.RangeKeys(func(key []byte, val []byte) bool {
+		numKeys++
+		return true
+	})
+
+	expNumKeys := 3
+	require.Equal(t, expNumKeys, numKeys)
+
+	err = db2.Close()
+	require.Nil(t, err)
+
+	err = db2.DestroyClosed()
+	require.Nil(t, err)
+
+}
+
+// ---------- Benchmarks ----------
+
+func BenchmarkPersisterOneKey1mil(b *testing.B) {
```

Comment:
can you move the benchmarks into a different file?
e.g shardedLeveldbSerial_bench_test.go

also I would remove the *OneKey* benchmarks as these take too long and the benchmark timeouts.

Reply:
done